### PR TITLE
SELC-937 contract upload notify with payload

### PR DESCRIPTION
**Pull request by antonioT90**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/partymanagement/service/impl/InstitutionServiceImpl.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+class InstitutionServiceImpl(
+  system: ActorSystem[_],
+  sharding: ClusterSharding,
+  entity: Entity[Command, ShardingEnvelope[Command]]
+) extends InstitutionService {
+
+  private val settings: ClusterShardingSettings = entity.settings match {
+    case None    => ClusterShardingSettings(system)
+    case Some(s) => s
+  }
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partymanagement/service/impl/InstitutionServiceImpl.scala]:
```suggestion
  private val settings: ClusterShardingSettings = entity.settings.getOrElse(ClusterShardingSettings(system))
```

File: src/main/scala/it/pagopa/interop/partymanagement/service/impl/RelationshipServiceImpl.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  system: ActorSystem[_],
+  sharding: ClusterSharding,
+  entity: Entity[Command, ShardingEnvelope[Command]]
+)(implicit ec: ExecutionContext)
+    extends RelationshipService {
+
+  private val settings: ClusterShardingSettings = entity.settings match {
+    case None    => ClusterShardingSettings(system)
+    case Some(s) => s
+  }
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/partymanagement/service/impl/RelationshipServiceImpl.scala]:
```suggestion
  private val settings: ClusterShardingSettings = entity.setting.getOrElse(ClusterShardingSettings(system))
```

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/partymanagement/model/persistence/PartyPersistentContractsProjection.scala [hunk 3]
```diff
@@ -1,80 +1,157 @@
...
-//        case Failure(e) =>
-//          logger.error(s"Error projecting confirm of relationship having id ${event.partyRelationshipId} on queue", e)
-//          DBIOAction.failed(e)
-//        case Success(_) =>
-//          DBIOAction.successful(Done)
-//      }
-//      future.as(Done)
-//    }
-//  }
-//}
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/partymanagement/model/persistence/PartyPersistentContractsProjection.scala:80]: What's the gain in having 80 lines of commented code in your code base? VSC exists for that.

File: src/main/scala/it/pagopa/interop/partymanagement/server/impl/Main.scala [hunk 4]
```diff
@@ -109,40 +124,56 @@ object Main extends App {
...
-//          new PartyPersistentContractsProjection(context.system, dbConfig, datalakeContractsPublisher)
-//
-//        ShardedDaemonProcess(context.system).init[ProjectionBehavior.Command](
-//          name = "party-contracts-projections",
-//          numberOfInstances = numberOfProjectionTags,
-//          behaviorFactory =
-//            (i: Int) => ProjectionBehavior(partyPersistentContractsProjection.projection(projectionTag(i))),
-//          stopMessage = ProjectionBehavior.Stop
-//        )
-//      }
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/partymanagement/server/impl/Main.scala:137]: ditto here

File: src/main/scala/it/pagopa/interop/partymanagement/model/persistence/PartyPersistentContractsProjection.scala (hunk 3 above)
Comment (antonioT90) [reply src/main/scala/it/pagopa/interop/partymanagement/model/persistence/PartyPersistentContractsProjection.scala:80]: I removed the comments :) It was previously commented in order to debug the dependencies error (Classpath not found...)

I just restored the code

File: src/main/scala/it/pagopa/interop/partymanagement/server/impl/Main.scala (hunk 4 above)
Comment (antonioT90) [reply src/main/scala/it/pagopa/interop/partymanagement/server/impl/Main.scala:137]: just restored the code configuring the projections, previously commented for debug purposes

**Review by galales — APPROVED** [review]

**Pull request merged**